examples/this-that/app/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+#![no_std]
+
+use gmeta::{InOut, Metadata};
+#[allow(unused_imports)]
+use gstd::debug;
+use gstd::prelude::*;
+use sails_macros::{command_handlers, query_handlers};
+
+pub struct ProgramMetadata;
+
+// TODO: This might be generated by some macro like `gprogram`
+impl Metadata for ProgramMetadata {
+    type Init = ();
+    type Handle = InOut<commands::Commands, commands::CommandResponses>; // Uses generated types
+    type Others = ();
+    type Reply = ();
+    type Signal = ();
+    type State = InOut<queries::Queries, queries::QueryResponses>; // Uses generated types
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub struct DoThatParam {
+    pub p1: u32,
+    pub p2: String,
+    pub p3: ManyVariants,
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub struct TupleStruct(bool);
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum ManyVariants {
+    One,
+    Two(u32),
+    Three(Option<u32>),
+    Four { a: u32, b: Option<u16> },
+    Five(String, u32),
+    Six((u32,)), // IDEA can't handle meta with single element tuple
+}
+
+#[command_handlers]
+pub mod commands {
+    use super::*;
+
+    // This
+    fn do_this(
+        p1: u32,
+        p2: String,
+        p3: (Option<String>, u8),
+        p4: TupleStruct,
+    ) -> Result<(String, u32), String> {
+        debug!("Handling 'do_this': {}, {}, {:?}, {:?}", p1, p2, p3, p4);
+        Ok((p2, p1))
+    }
+
+    // That
+    fn do_that(
+        param: DoThatParam,
+    ) -> Result<(String, u32), (String,) /* IDEA can't handle meta with single element tuple */>

```

Comment:
why do you need here single element tuple anyway?

Reply:
This is just as many possible type usages as I could imagine.